torchgen/gen.py

```diff
@@ -2661,13 +2663,15 @@ def gen_op_headers(
         key_fn=key_func,
         env_callable=functionalization_env_callable,
         num_shards=4,
-        sharded_keys={
-            "ops_headers",
-            "func_definitions",
-            "func_registrations",
-            "func_add_back_views_definitions",
-            "func_add_back_views_registrations",
-        },
+        sharded_keys=frozenset(
```

Comment:
Technically, only the default args need to be frozenset, non-default args can be normal sets.

Reply:
whoops! I'll see what the minimal change is then

Reply:
Yeah, you just don't want the default args to be accidentally mutated since they are singletons.